src/lib/core.d.ts

```diff
@@ -230,7 +232,8 @@ interface Function {
       * @param thisArg An object to which the this keyword can refer inside the new function.
       * @param argArray A list of arguments to be passed to the new function.
       */
-    bind(thisArg: any, ...argArray: any[]): any;
+    bind<T, U>(this: (this: T, ...argArray: any[]) => U, thisArg: T, ...argArray: any[]): (this: void, ...argArray: any[]) => U;

```

Comment:
`T` is the function's `this` type. `U` is the function's return value. There's no relation between them. I don't know TypeScript's conventions for type variable names that well. Would it be easier to read to change this to `<TThis, TReturn>`?